agent/__init__.py

```diff
@@ -209,28 +209,102 @@ def get_ca_cert(debug=False):
     return ca.json()['ca_bundle']
 
 
+def get_mtls_header(debug=False, dev=False):
+    return {
+        'SSL-CLIENT-SUBJECT-DN': 'CN=' + get_device_id(debug=debug, dev=dev),
+        'SSL-CLIENT-VERIFY': 'SUCCESS'
+    } if dev else {}
+
+
+def mtls_req_error_log(request_url, req_type, requester_name, response):
+    """
+    logs error of mtls_request functions
+    :param request_url: request url string
+    :param req_type: 'GET', 'POST', ...
+    :param requester_name: requester id for message, if None then request_url string used
+    :param response - request response
+    :return: None
+    """
+    if not requester_name:
+        requester_name = "({})".format(request_url)
+    print("wott-agent :: mtls_post_request :: [RECEIVED] {} {}: {}".format(
+        requester_name, req_type, response.status_code))
+    print("wott-agent :: mtls_post_request :: [RECEIVED] {} {}: {}".format(
+        requester_name, req_type, response.content))
+
+
+def mtls_post_request(req, data=None, debug=False, dev=False, requester_name=None, debug_on_ok=False):
+    """
+    MTLS POST Request wrapper function.
+    :param req: request url string (without endpoint)
+    :param data: json data, None by default
+    :param debug: if true then log error messages
+    :param dev: if true use dev endpoint and dev headers
+    :param requester_name: displayed requester id for error messages
+    :param debug_on_ok: if true with debug then log successful response status/content too
+    :return: response or None (if there was exception raised)
+    """
+    try:
+        r = requests.post(
+            '{}/v0.2/{}'.format(MTLS_ENDPOINT, req),
+            cert=(CLIENT_CERT_PATH, CLIENT_KEY_PATH),
+            json=data,
+            headers=get_mtls_header(debug=debug, dev=dev)
+        )
+
+        if (debug_on_ok or not r.ok) and debug:
+            mtls_req_error_log(req, 'POST', requester_name, r)
+        return r
+
+    except Exception as e:
+        print('wott-agent :: mtls_post_request :: rises exception: {}'.format(e))
+        return None
+
+
+def mtls_get_request(req, debug=False, dev=False, requester_name=None, debug_on_ok=False):
+    """
+    MTLS GET Request wrapper function.
+    :param req: request url string (without endpoint)
+    :param debug: if true then log error messages
+    :param dev: if true use dev endpoint and dev headers
+    :param requester_name: displayed requester id for error messages
+    :param debug_on_ok: if true with debug then log successful response status/content too
+    :return: response or None (if there was exception raised)
+    """
+    try:
+        r = requests.get(
+            '{}/v0.2/{}'.format(MTLS_ENDPOINT, req),
+            cert=(CLIENT_CERT_PATH, CLIENT_KEY_PATH),
+            headers=get_mtls_header(debug=debug, dev=dev)
+        )
+
+        if (debug_on_ok or not r.ok) and debug:
+            mtls_req_error_log(req, 'GET', requester_name, r)
+        return r
+
+    except Exception as e:
```

Comment:
Swallowing all exceptions is not a good idea. I wouldn't do it at all here.

Reply:
In any network problems agent clogs the screen with exception and hungs if not in daemon mode.  It not looks good. The only one function where it was handled some way is `get_claim_token()` but that handle only `connection error` exception and usethe same message for bad status and exception.  So i chnage whole exceptions to only requests one and left the way to get it as second result for future use.

Reply:
moreover we plan to refactor logging from print to logging. then it would be better to remove that exception return of mine and logs unhandled exceptions as errors